File: process_youtube.js
const fs = require('fs');

try {
    const raw = JSON.parse(fs.readFileSync('data/youtube_raw.json', 'utf8'));

    if (!raw.items) {
        console.error('No items found in raw data');
        process.exit(1);
    }

    const videos = raw.items.map(item => ({
        id: item.snippet.resourceId.videoId,
        title: item.snippet.title,
        thumbnail: item.snippet.thumbnails.medium?.url || item.snippet.thumbnails.default?.url,
        publishedAt: item.snippet.publishedAt
    }));

    fs.writeFileSync('data/youtube.json', JSON.stringify(videos, null, 2));
    console.log(`Saved ${videos.length} videos to data/youtube.json`);
} catch (e) {
    console.error(e);
}

File: process_youtube.py
import json

try:
    with open('data/youtube_raw.json', 'r', encoding='utf-8') as f:
        raw = json.load(f)
    
    videos = []
    if 'items' in raw:
        for item in raw['items']:
            snippet = item['snippet']
            thumbnails = snippet.get('thumbnails', {})
            thumb = thumbnails.get('medium', {}).get('url') or thumbnails.get('default', {}).get('url')
            
            videos.append({
                'id': snippet['resourceId']['videoId'],
                'title': snippet['title'],
                'thumbnail': thumb,
                'publishedAt': snippet['publishedAt']
            })
            
    with open('data/youtube.json', 'w', encoding='utf-8') as f:
        json.dump(videos, f, indent=2, ensure_ascii=False)
        
    print(f"Saved {len(videos)} videos to data/youtube.json")

except Exception as e:
    print(f"Error: {e}")
